src/query/frontend/ast/cypher_main_visitor.cpp

```diff
@@ -492,47 +496,281 @@ antlrcpp::Any CypherMainVisitor::visitStreamQuery(MemgraphCypher::StreamQueryCon
+      LOG_FATAL("Invalid config key used");
+  }
+}
+
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#define GENERATE_STREAM_CONFIG_KEY_ENUM(stream, first_config, ...)   \
+  enum class BOOST_PP_CAT(stream, ConfigKey) : uint8_t {             \
+    first_config = static_cast<uint8_t>(CommonStreamConfigKey::END), \
+    __VA_ARGS__                                                      \
+  };
+
+GENERATE_STREAM_CONFIG_KEY_ENUM(Kafka, TOPICS, CONSUMER_GROUP, BOOTSTRAP_SERVERS);
+
+constexpr std::array all_kafka_config_keys{KafkaConfigKey::TOPICS, KafkaConfigKey::CONSUMER_GROUP,
+                                           KafkaConfigKey::BOOTSTRAP_SERVERS};
+
+std::string_view ToString(const KafkaConfigKey key) {
+  switch (key) {
+    case KafkaConfigKey::TOPICS:
+      return "TOPICS";
+    case KafkaConfigKey::CONSUMER_GROUP:
+      return "CONSUMER_GROUP";
+    case KafkaConfigKey::BOOTSTRAP_SERVERS:
+      return "BOOTSTRAP_SERVERS";
+  }
+}
+
+void MapCommonStreamConfigs(auto &memory, StreamQuery &stream_query) {
+  for (const auto key : all_common_stream_config_keys) {
+    switch (key) {
+      case CommonStreamConfigKey::TRANSFORM:
+        MapConfig<true, std::string>(memory, CommonStreamConfigKey::TRANSFORM, stream_query.transform_name_);
+        break;
+      case CommonStreamConfigKey::BATCH_INTERVAL:
+        MapConfig<false, Expression *>(memory, CommonStreamConfigKey::BATCH_INTERVAL, stream_query.batch_interval_);
+        break;
+      case CommonStreamConfigKey::BATCH_SIZE:
+        MapConfig<false, Expression *>(memory, CommonStreamConfigKey::BATCH_SIZE, stream_query.batch_size_);
+        break;
+      case CommonStreamConfigKey::END:
+        LOG_FATAL("Invalid config key used");
+    }
+  }
+}
+}  // namespace
+
+antlrcpp::Any CypherMainVisitor::visitKafkaCreateStream(MemgraphCypher::KafkaCreateStreamContext *ctx) {
   auto *stream_query = storage_->Create<StreamQuery>();
   stream_query->action_ = StreamQuery::Action::CREATE_STREAM;
+  stream_query->type_ = StreamQuery::Type::KAFKA;
   stream_query->stream_name_ = ctx->streamName()->symbolicName()->accept(this).as<std::string>();
 
-  auto *topic_names_ctx = ctx->topicNames();
+  for (auto *create_config_ctx : ctx->kafkaCreateStreamConfig()) {
+    create_config_ctx->accept(this);
+  }
+
+  for (const auto key : all_kafka_config_keys) {
+    switch (key) {
+      case KafkaConfigKey::TOPICS:
+        MapConfig<true, std::vector<std::string>, Expression *>(memory_, KafkaConfigKey::TOPICS,
+                                                                stream_query->topic_names_);
+        break;
+      case KafkaConfigKey::CONSUMER_GROUP:
+        MapConfig<false, std::string>(memory_, KafkaConfigKey::CONSUMER_GROUP, stream_query->consumer_group_);
+        break;
+      case KafkaConfigKey::BOOTSTRAP_SERVERS:
+        MapConfig<false, Expression *>(memory_, KafkaConfigKey::BOOTSTRAP_SERVERS, stream_query->bootstrap_servers_);
+        break;
+    }
+  }
```

Comment:
Because I want that compile error when a new Config is added that is basically "hey don't forget to change this". Forcing a switch statement seemed the easiest way to do it.

Reply:
This can be also caught with tests so maybe it's an overkill :smile:

Reply:
I think with this solution you have to not to forget to add the key to `all_kafka_config_keys` array, which is right below the enum itself, so it is hard to forget, that is true. On the other side, I think when you add a new config key, then you would use it only here, with a `MapConfig` call. therefore I think it is hard to forget it. The scope of these enums are very narrow (only here, if I am not mistaken), not scattered around the whole codebase, thus I would say I don't see the value in compile error. But it is fine for me either way.

Reply:
I see, I'll remove it, you're right. Also, it's easy to catch with tests as I said.